pypeit/core/telluric.py

```diff
@@ -2263,8 +2447,23 @@ def __init__(self, wave, flux, ivar, gpm, telgridfile, obj_params, init_obj_mode
                 wave, flux, ivar, gpm)
         # 3) Read the telluric grid and initalize associated parameters
         wv_gpm = self.wave_in_arr > 1.0
-        self.tell_dict = read_telluric_grid(self.telgrid, wave_min=self.wave_in_arr[wv_gpm].min(),
-                                            wave_max=self.wave_in_arr[wv_gpm].max())
+        if teltype == 'pca' or teltype == 'PCA':
```

Comment:
Can you instead wrap the reading functions into general readers and evaluators that take teletype as an argument. Then we could get rid of this relatively ugly if then statement.

Reply:
I wrapped the evaluators into one function since that makes sense, but left the readers separate, as they output dramatically different dicts